src/login.ts

```diff
@@ -1,57 +1,92 @@
-import type { RequestInit } from '@mastojs/ponyfills';
 import { SemVer } from 'semver';
 
 import type { MastoConfigProps } from './config';
 import { MastoConfig } from './config';
 import { HttpNativeImpl } from './http';
-import type { LogType } from './logger';
+import type { Logger } from './logger';
 import { LoggerConsoleImpl } from './logger';
+import type { v1, v2 } from './mastodon';
 import { Client } from './mastodon';
-import { InstanceRepository } from './mastodon/v1/repositories';
+import { InstanceRepository as V1InstanceRepository } from './mastodon/v1/repositories';
+import { InstanceRepository as V2InstanceRepository } from './mastodon/v2/repositories';
 import { SerializerNativeImpl } from './serializers';
 import type { Writable } from './utils';
 import { WsNativeImpl } from './ws';
 
-export type LoginParams = {
-  /** URL of your instance */
-  readonly url: string;
-  /** Log level to print to your console */
-  readonly logLevel?: LogType;
-  /** Access token of your account */
-  readonly accessToken?: string;
-  /** Timeout in milliseconds */
-  readonly timeout?: number;
-  readonly defaultRequestInit?: Omit<RequestInit, 'body' | 'method'>;
-  readonly disableVersionCheck?: boolean;
-  readonly disableDeprecatedWarning?: boolean;
+export type RequestParams = Pick<
+  MastoConfigProps,
+  | 'url'
+  | 'logLevel'
+  | 'timeout'
+  | 'defaultRequestInit'
+  | 'disableDeprecatedWarning'
+>;
+
+export type LoginParams = Omit<MastoConfigProps, 'version'> & {
```

Comment:
@sxzz Just one more thing: I think this should be named `CreateClientParams` and change `LoginParam` to `Omit<CreateClientParams, 'streamingApiUrl' | 'version'>` instead because it's natural to correspond the function name to its parameter type name.
Let me know by hitting the re-request button when you're ready to merge this PR 🙏